src/patternfly/components/Menu/menu-item.hbs

```diff
@@ -1,17 +1,23 @@
-<{{# if menu-item--IsLink}}a{{else if menu-item--type}}{{menu-item--type}}{{else}}button{{/if}} class="pf-c-menu__item{{#if menu-item--modifier}} {{menu-item--modifier}}{{/if}}{{#if menu-list-item--IsCurrent}} pf-m-current{{/if}}"
-  {{#if menu-item--IsLink}}
-    href="#"
-    {{#if menu-list-item--IsDisabled}}
-      aria-disabled="true"
-      tabindex="-1"
-    {{/if}}
-  {{else}}
-    type="button"
-    {{#if menu-list-item--IsDisabled}}
-      disabled
+<{{#if menu-item--IsLink}}a{{else if menu-item--type}}{{menu-item--type}}{{else if menu-list-item--IsCheckbox}}label{{else}}button{{/if}} class="pf-c-menu__item{{#if menu-item--modifier}} {{menu-item--modifier}}{{/if}}
+  {{~#if menu-list-item--IsCurrent}} pf-m-current{{/if}}"
+  {{#unless menu-list-item--IsCheckbox}}
+    {{#if menu-item--IsLink}}
+      href="#"
+      {{#if menu-list-item--IsDisabled}}
+        aria-disabled="true"
+        tabindex="-1"
+      {{/if}}
+    {{else}}
+      type="button"
+      {{#if menu-list-item--IsDisabled}}
+        disabled
+      {{/if}}
     {{/if}}
+    role="menuitem"
+  {{/unless}}
+  {{#if menu-list-item--IsCheckbox}}
```

Comment:
just a nit - if you do `{{#unless menu-list-item--IsCheckbox}}...{{/unless}}{{#if menu-list-item--IsCheckbox}}...{{/if}}`, that could be rewritten as `{{#if menu-list-item--IsCheckbox}}...{{else}}...{{/if}}`